### ISSUE TITLE
Parameter inside dictionary not resolved

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#79]
JSON input:

```
{
   "val01" : "string",
   "val02" : 123,
   "val03" : "${val01}",
   "val04" : ["A", "B", "C"],
   "val05" : {"A" : "${val01}", "B" : "${val02}", "C" : "${val03}"},
   "val06" : true,
   "val07" : false,
   "val08" : null
}
```
JsonPreprocessor output:

```
{'val01': 'string',
 'val02': 123,
 'val03': 'string',
 'val04': ['A', 'B', 'C'],
 'val05': {'A': 'string', 'B': 123, 'C': '${val01}'},
 'val06': True,
 'val07': False,
 'val08': None}
```

The value of `val05.C` is `'${val01}'`, even in case of the value of `val03` is already `'string'`.

Expected: The value of `val05.C` is `'string'`.

**Comment by test-fullautomation** [comment]
Hi Son,
please look into this.
Expectation is what Holger writes.
Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Holger, Hello Thomas,

This scenario is similar with the scenario Holger mentioned in issue #80 that a nested parameter is used to other nested parameter.
In this case, the parameter val03 is defined with nested param "${val01}", then "${val03}" is used as nested param in val05.

My concern again is: Dose this scenario occurs in reality? 

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Son,
yes, the preprocessor must resolve this properly, too. Any scenario is possible.
Parameters must be resolved according to their definition order recursively.
Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Holger, Hello Thomas,

This issue was fixed with in [commit e6851712](https://github.com/test-fullautomation/python-jsonpreprocessor/commit/e685171279232cdde9cbbb99b14df9f481fa93d4) on the stabi branch ([namsonx/task/stabi_branch](https://github.com/test-fullautomation/python-jsonpreprocessor/tree/namsonx/task/stabi_branch))

Thank you,
Son

**Comment by HolQue** [comment]
Retest successful

**Comment by test-fullautomation** [comment]
solved and already merged #147
